indexer/services/comlink/src/controllers/api/v4/addresses-controller.ts

```diff
@@ -241,6 +241,107 @@ class AddressesController extends Controller {
     );
     return subaccountResponse;
   }
+
+  @Get('/:address/parentSubaccountNumber/:parentSubaccountNumber')
+  public async getParentSubaccount(
+    @Path() address: string,
+      @Path() parentSubaccountNumber: number,
+  ): Promise<ParentSubaccountResponse> {
+
+    const childSubaccountIds: string[] = getChildSubaccountNums(parentSubaccountNumber).map(
+      (subaccountNumber: number): string => SubaccountTable.uuid(address, subaccountNumber),
+    );
+
+    // TODO(IND-189): Use a transaction across all the DB queries
+    const [subaccounts, latestBlock]: [
+      SubaccountFromDatabase[],
+      BlockFromDatabase,
+    ] = await Promise.all([
+      SubaccountTable.findAll(
+        {
+          id: childSubaccountIds,
+          address,
+        },
+        [],
+      ),
+      BlockTable.getLatest(),
+    ]);
+
+    if (subaccounts.length === 0) {
+      throw new NotFoundError(`No subaccounts found for address ${address} and parentSubaccountNumber ${parentSubaccountNumber}`);
+    }
+
+    const latestFundingIndexMap: FundingIndexMap = await FundingIndexUpdatesTable
+      .findFundingIndexMap(
+        latestBlock.blockHeight,
+      );
+
+    const subaccountResponses: SubaccountResponseObject[] = await Promise.all(subaccounts.map(
+      async (subaccount: SubaccountFromDatabase): Promise<SubaccountResponseObject> => {
+        const [
+          perpetualPositions,
+          assetPositions,
+          assets,
+          markets,
+          lastUpdatedFundingIndexMap,
+        ] = await Promise.all([
+          getOpenPerpetualPositionsForSubaccount(
+            subaccount.id,
+          ),
+          getAssetPositionsForSubaccount(
+            subaccount.id,
+          ),
```

Comment:
Can we pull these values in bulk? I'm worried this will take too much time?

Reply:
I think this is a valid concern. There's a lot of ways to improve performance overall in all these APIs. My thinking right now is lets get these changes out first. If the performance is good enough, we don't need to tackle it immediately. If its not great, we can look into ways to improve perf

Reply:
that sounds good for now, we do need to monitor this, can you make sure the metric for this PR would be in v4-infrastructure and in our datadog for metrics?

Reply:
should be available with the new commit